### PR TITLE
Origin/13.0

### PR DESCRIPTION
**Pull request by tonybenoy**

### CODE REVIEW

**Comment by tonybenoy** [comment]
@Yenthe666 Check this out.  I was occupied so couldn't update you.

File: odoo_install.sh [hunk 1]
```diff
@@ -90,7 +97,7 @@ if [ $INSTALL_WKHTMLTOPDF = "True" ]; then
   sudo ln -s /usr/local/bin/wkhtmltopdf /usr/bin
   sudo ln -s /usr/local/bin/wkhtmltoimage /usr/bin
 else
-  echo "Wkhtmltopdf isn't installed due to the choice of the user!"
+  echo "Wkhtmltopdf isn't installed due to choice of the user!"
```
Comment (Yenthe666) [inline odoo_install.sh]: can you please change this term back so its not modified?

File: odoo_install.sh [hunk 2]
```diff
@@ -17,30 +17,37 @@
...
+# Set the default Odoo longpolling port (you still have to use -c /etc/odoo-server.conf for example to use this.)
+LONGPOLLING_PORT="8072"
 
 ##
 ###  WKHTMLTOPDF download links
 ## === Ubuntu Trusty x64 & x32 === (for other distributions please replace these two links,
-## in order to have correct version of wkhtmltopdf installed, for a danger note refer to 
-## https://github.com/odoo/odoo/wiki/Wkhtmltopdf ):
+## in order to have correct version of wkhtmltox installed, for a danger note refer to
+## https://www.odoo.com/documentation/8.0/setup/install.html#deb ):
```
Comment (Yenthe666) [inline odoo_install.sh]: Let's remove this line

File: odoo_install.sh [hunk 3]
```diff
@@ -112,8 +119,7 @@ if [ $IS_ENTERPRISE = "True" ]; then
     # Odoo Enterprise install!
     echo -e "\n--- Create symlink for node"
     sudo ln -s /usr/bin/nodejs /usr/bin/node
-    sudo su $OE_USER -c "mkdir $OE_HOME/enterprise"
-    sudo su $OE_USER -c "mkdir $OE_HOME/enterprise/addons"
+    sudo su $OE_USER -c "mkdir -p $OE_HOME/enterprise/addons"
```
Comment (Yenthe666) [inline odoo_install.sh]: Let's not change this in a stable version for sure.

File: odoo_install.sh [hunk 4]
```diff
@@ -134,14 +140,14 @@ if [ $IS_ENTERPRISE = "True" ]; then
...
-sudo su $OE_USER -c "mkdir $OE_HOME/custom"
-sudo su $OE_USER -c "mkdir $OE_HOME/custom/addons"
+sudo su $OE_USER -c "mkdir -p $OE_HOME/custom/addons"
 
 echo -e "\n---- Setting permissions on home folder ----"
 sudo chown -R $OE_USER:$OE_USER $OE_HOME/*
 
 echo -e "* Create server config file"
 
+
```
Comment (Yenthe666) [inline odoo_install.sh:152]: No need for the diff?

File: odoo_install.sh [hunk 5]
```diff
@@ -150,8 +156,13 @@ if [ $GENERATE_RANDOM_PASSWORD = "True" ]; then
     OE_SUPERADMIN=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 16 | head -n 1)
 fi
 sudo su root -c "printf 'admin_passwd = ${OE_SUPERADMIN}\n' >> /etc/${OE_CONFIG}.conf"
-sudo su root -c "printf 'xmlrpc_port = ${OE_PORT}\n' >> /etc/${OE_CONFIG}.conf"
+if [ $OE_VERSION == "12.0" ];then
```
Comment (Yenthe666) [inline odoo_install.sh]: Hmm, not sure why we should do this?

File: odoo_install.sh [hunk 6]
```diff
@@ -244,6 +255,88 @@ sudo chown root: /etc/init.d/$OE_CONFIG
...
+   listen 80;
+
+   # set proper server name after domain set
+   server_name $WEBSITE_NAME;
+
+   # Add Headers for odoo proxy mode
+   proxy_set_header X-Forwarded-Host \$host;
+   proxy_set_header X-Forwarded-For \$proxy_add_x_forwarded_for;
+   proxy_set_header X-Forwarded-Proto \$scheme;
+   proxy_set_header X-Real-IP \$remote_addr;
```
Comment (Yenthe666) [inline odoo_install.sh]: Maybe we should add these two too?
 proxy_set_header X-Client-IP $remote_addr;
 proxy_set_header HTTP_X_FORWARDED_HOST $remote_addr;

File: odoo_install.sh [hunk 7]
```diff
@@ -244,6 +255,88 @@ sudo chown root: /etc/init.d/$OE_CONFIG
...
+
+   #   odoo    log files
+   access_log  /var/log/nginx/$OE_USER-access.log;
+   error_log       /var/log/nginx/$OE_USER-error.log;
+
+   #   increase    proxy   buffer  size
+   proxy_buffers   16  64k;
+   proxy_buffer_size   128k;
+
+   proxy_read_timeout 720s;
```
Comment (Yenthe666) [inline odoo_install.sh]: I'd bump these up higher by default, its pretty short?

File: odoo_install.sh [hunk 8]
```diff
@@ -244,6 +255,88 @@ sudo chown root: /etc/init.d/$OE_CONFIG
...
+   gzip    on;
+   gzip_min_length 1100;
+   gzip_buffers    4   32k;
+   gzip_types  text/css text/less text/plain text/xml application/xml application/json application/javascript application/pdf image/jpeg image/png;
+   gzip_vary   on;
+   client_header_buffer_size 4k;
+   large_client_header_buffers 4 64k;
+   client_max_body_size 0;
+
+   location / {
```
Comment (Yenthe666) [inline odoo_install.sh]: Maybe we should also add
 location ~* .(js|css|png|jpg|jpeg|gif|ico)$ {
   expires 2d;
   proxy_pass http://odoo;
   add_header Cache-Control "public, no-transform";
 }

File: odoo_install.sh (hunk 5 above)
Comment (tonybenoy) [reply odoo_install.sh]: If I remember correctly the xmlrpc port  conf was changed after 12 right? Anyways I've changed it to >=12.0

**Comment by tonybenoy** [comment]
@Yenthe666 Let me know if there are any changes needed

File: odoo_install.sh [hunk 9]
```diff
@@ -17,30 +17,37 @@
 OE_USER="odoo"
 OE_HOME="/$OE_USER"
 OE_HOME_EXT="/$OE_USER/${OE_USER}-server"
-# The default port where this Odoo instance will run under (provided you use the command -c in the terminal)
-# Set to true if you want to install it, false if you don't need it or have it already installed.
+#The default port where this Odoo instance will run under (provided you use the command -c in the terminal)
+#Set to true if you want to install it, false if you don't need it or have it already installed.
 INSTALL_WKHTMLTOPDF="True"
-# Set the default Odoo port (you still have to use -c /etc/odoo-server.conf for example to use this.)
+#Set the default Odoo port (you still have to use -c /etc/odoo-server.conf for example to use this.)
```
Comment (Yenthe666) [inline odoo_install.sh]: can you please add back the space to keep the diff minimal?

File: odoo_install.sh [hunk 10]
```diff
@@ -17,30 +17,37 @@
 OE_USER="odoo"
 OE_HOME="/$OE_USER"
 OE_HOME_EXT="/$OE_USER/${OE_USER}-server"
-# The default port where this Odoo instance will run under (provided you use the command -c in the terminal)
-# Set to true if you want to install it, false if you don't need it or have it already installed.
+#The default port where this Odoo instance will run under (provided you use the command -c in the terminal)
+#Set to true if you want to install it, false if you don't need it or have it already installed.
```
Comment (Yenthe666) [inline odoo_install.sh]: can you please add back the space to keep the diff minimal?

**Comment by Yenthe666** [comment]
I think that after this minor change we've got things into place :)

**Comment by tonybenoy** [comment]
@Yenthe666 Done

File: odoo_install.sh [hunk 11]
```diff
@@ -244,6 +257,94 @@ sudo chown root: /etc/init.d/$OE_CONFIG
...
+   proxy_buffering    on;
+   expires 864000;
+   proxy_pass    http://127.0.0.1:$OE_PORT;
+   }
+   }
+EOF
+
+  sudo mv ~/$OE_USER /etc/nginx/sites-available/
+  sudo ln -s /etc/nginx/sites-available/odoo /etc/nginx/sites-enabled/$OE_USER
+  sudo rm /etc/nginx/sites-enabled/default
```
Comment (Yenthe666) [inline odoo_install.sh:341]: Should we really do this though?

File: odoo_install.sh [hunk 12]
```diff
@@ -244,6 +257,94 @@ sudo chown root: /etc/init.d/$OE_CONFIG
 echo -e "* Start ODOO on Startup"
 sudo update-rc.d $OE_CONFIG defaults
 
+#--------------------------------------------------
+# Install Nginx if needed
+#--------------------------------------------------
+if [ $INSTALL_NGINX = "True" ]; then
+  echo -e "\n---- Installing and setting up Nginx ----"
+  sudo apt install nginx -y
+  cat <<EOF > ~/$OE_USER
```
Comment (Yenthe666) [inline odoo_install.sh]: Could you remove those odd tabs and replace them by nice spaces & also indent the blocks neatly? If we have this I'll merge it in ;)

**Comment by Yenthe666** [comment]
@tonybenoy any chance you can do the final request? 🙏

**Comment by tonybenoy** [comment]
@Yenthe666 done :+1:

**Comment by Yenthe666** [comment]
Merged in at https://github.com/Yenthe666/InstallScript/commit/775ee313fa197059e9cd3a87e5d3ab2718577294
Thank you very much @tonybenoy! 👊 👍

**Pull request merged**